x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/225_query_ruleset_list.yml

```diff
@@ -0,0 +1,161 @@
+                  value: elastic
+              actions:
+                ids:
+                  - 'id1'
+                  - 'id2'
+            - rule_id: query-rule-id2
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: kibana
+              actions:
+                ids:
+                  - 'id3'
+                  - 'id4'
+
+  - do:
+      query_ruleset.put:
+        ruleset_id: test-query-ruleset-2
+        body:
+          ruleset_id: test-query-ruleset-2
+          rules:
+            - rule_id: query-rule-id1
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: elastic
+              actions:
+                ids:
+                  - 'id1'
+                  - 'id2'
+            - rule_id: query-rule-id2
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: kibana
+              actions:
+                ids:
+                  - 'id3'
+                  - 'id4'
+            - rule_id: query-rule-id3
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: logstash
+              actions:
+                ids:
+                  - 'id5'
+                  - 'id6'
+
+  - do:
+      query_ruleset.put:
+        ruleset_id: test-query-ruleset-3
+        body:
+          ruleset_id: test-query-ruleset-3
+          rules:
+            - rule_id: query-rule-id1
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: elastic
+              actions:
+                ids:
+                  - 'id1'
+                  - 'id2'
+            - rule_id: query-rule-id2
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: kibana
+              actions:
+                ids:
+                  - 'id3'
+                  - 'id4'
+            - rule_id: query-rule-id3
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: logstash
+              actions:
+                ids:
+                  - 'id5'
+                  - 'id6'
+            - rule_id: query-rule-id4
+              type: pinned
+              criteria:
+                - type: exact
+                  metadata: query_string
+                  value: beats
+              actions:
+                ids:
+                  - 'id7'
+                  - 'id8'
+
+---
+"List Query Rulesets":
+  - do:
+      query_ruleset.list: { }
+
+  - match: { count: 3 }
+
+  # Alphabetical order by ruleset_id for results
```

Comment:
👍 .  We probably could insert them in other order in the `setup`, to ensure the order is alphabetical

Reply:
Nice suggestion, I'll change that!